src/studio/src/designer/frontend/packages/schema-editor/src/components/RestrictionField.tsx

```diff
@@ -40,18 +41,27 @@ const useStyles = makeStyles({
 
 export const RestrictionField = (props: IRestrictionFieldProps) => {
   const classes = useStyles();
-  const [val, setVal] = React.useState(props.value);
+  const [key, setKey] = React.useState(props.keyName);
   React.useEffect(() => {
-    setVal(props.value);
-  }, [props.value]);
+    setKey(props.keyName);
```

Comment:
I think this value could be called 'keyName' or anything else in this context.
'key' alone is a bit problematic to use, as it's hard to isolate from other usages of 'key'.
As key is also a reserved prop i React, I have to pause and double-check when I see usage of 'key'.
Ambiguity can quickly cause issues.

Reply:
Done